src/cpp/src/text_callback_streamer.hpp

```diff
@@ -9,14 +9,17 @@
 namespace ov {
 namespace genai {
 
-class TextCallbackStreamer: public StreamerBase {
+class TextCallbackStreamer : public StreamerBase {
 public:
+    TextCallbackStreamer(const Tokenizer& tokenizer, std::function<bool(std::string)> callback);
+
     bool put(int64_t token) override;
+    bool put(const std::vector<int64_t>& tokens) override;
     void end() override;
 
-    TextCallbackStreamer(const Tokenizer& tokenizer, std::function<bool(std::string)> callback);
-
-    std::function<bool(std::string)> on_finalized_subword_callback = [](std::string words)->bool { return false; };
+    std::function<bool(std::string)> on_finalized_subword_callback = [](std::string words) -> bool {
```

Comment:
should this field be moved to `protected` or even `private` section?

Reply:
Makes sense to me as well. @pavel-esir is there a reason for callback to be public? Don't you mind to move it to `private`?

Reply:
I suppose nobody cared about it, as class is `private` (now)

Reply:
Move to private